### PR TITLE
[ENH] Expose dropout parameters in CNTCNetwork and TapNetNetwork 

### PR DESCRIPTION
**Pull request by xenonnn4w**
#### Reference Issues/PRs
Fixes 
* #9103. 

Related to
*  #9109.

#### What does this implement/fix? Explain your changes.
This PR continues the work from #9109 to expose hidden layer dropout parameters in deep learning models. It addresses the remaining networks that had hardcoded dropout values:

#### CNTCNetwork:

Added dropout parameter that accepts either a float `uniform dropout` or a tuple of 7 floats `layer-wise dropout`
Default: `0.2, 0.2, 0.1, 0.1, 0.1, 0.1, 0.1` - matching the previously hardcoded values
The 7 values correspond to: conv1, rnn1, conv2, lstm, avg, attention, and mlp layers
Added validation to ensure tuple length matches expected layers

#### TapNetNetwork:

Added lstm_dropout parameter `default: 0.8` to expose the previously hardcoded LSTM dropout
The existing dropout parameter was not being used for the LSTM layer

#### Updated classifiers/regressors:

CNTCClassifier, CNTCRegressor: Added dropout parameter
TapNetClassifier, TapNetRegressor: Added lstm_dropout parameter
All changes are backward compatible - default values preserve existing behavior.

#### Any other comments?
The ConvTimeNet files `_dlutils.py, _patch_layers.py` have `hardcoded nn.Dropout(0.1)` but these are defined and never used in their forward() methods.

##### For all contributions

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

File: sktime/networks/tapnet.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ class TapNetNetwork(BaseDeepNetwork):
         parameters for random permutation
     dropout : float, default = 0.5
         dropout rate, in the range [0, 1)
+    lstm_dropout : float, default = 0.8
```
Comment (fkiraly) [inline sktime/networks/tapnet.py:33]: why is this separate from `droupout`?

**Review by fkiraly — CHANGES_REQUESTED** [review]
great, thanks!

Can you explain why in TapNet you introduced a new parameter, while there already was a `dropout` parameter?

File: sktime/networks/tapnet.py (hunk 1 above)
Comment (xenonnn4w) [reply sktime/networks/tapnet.py:33]: The reason I added a separate `lstm_dropout` parameter because:
#### Current situation in TapNetNetwork:
- The existing dropout parameter (default 0.5) exists but was not being used for the LSTM layer
- The LSTM layer had a hardcoded `Dropout(0.8)` ; a different value
#### My reasoning:
To maintain backward compatibility, I kept them separate so the default behavior doesn't change:
- `dropout=0.5` for other parts of the network
- `lstm_dropout=0.8` for the LSTM layer (matching the previously hardcoded value)

Comment (fkiraly) [reply sktime/networks/tapnet.py:33]: Makes sense!

**Review by fkiraly — APPROVED** [review]
